screens/ReservationCalendarScreen.js

```diff
@@ -0,0 +1,508 @@
+
+        this.getUsers()
+        this.props.getTableLayouts()
+        this.getReservationEvents()
+        this.getReservationEventsByDate()
+    }
+
+    refreshScreen = async () => {
+
+        this.getUsers()
+        this.props.getTableLayouts()
+        this.getReservationEvents(new Date(this.state?.selectedDate ?? new Date()).getFullYear(), moment(new Date(this.state?.selectedDate ?? new Date())).format('MM'))
+        this.getReservationEventsByDate(this.state?.selectedDate, 'BOOKED')
+    }
+
+    getReservationEvents = async (year = new Date().getFullYear(), month = moment(new Date()).format('MM')) => {
+
+        this.setState({isLoading: true})
+
+        await dispatchFetchRequest(api.reservation.getReservationByMonth(year, month), {
+            method: 'GET',
+            withCredentials: true,
+            credentials: 'include',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+        }, response => {
+            response.json().then(async (data) => {
+                this.setState({reservationEvents: data?.results})
+                this.setState({
+                    isLoading: false, nowYear: new Date(this.state?.selectedDate).getFullYear(),
+                    nowMonth: new Date(this.state?.selectedDate).getMonth() + 1
+                })
+            })
+        }).then().catch(() => this.setState({isLoading: false}))
+    }
+
+    getReservationEventsByDate = async (date = this.state.selectedDate, status = 'BOOKED') => {
+
+        await dispatchFetchRequest(api.reservation.getReservationByDate(date, status), {
+            method: 'GET',
+            withCredentials: true,
+            credentials: 'include',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+        }, response => {
+            response.json().then(async (data) => {
+                this.setState({reservationEventsByDate: data?.results})
+            })
+        }).then(
+            await dispatchFetchRequest(api.reservation.getReservationByDate(date, 'CONFIRMED'), {
+                method: 'GET',
+                withCredentials: true,
+                credentials: 'include',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+            }, response => {
+                response.json().then(async (data) => {
+                    this.setState({reservationEventsByDate: [...this.state.reservationEventsByDate, data?.results]})
+                })
+            }).then()
+        )
+
+
+    }
+
+
+
+    handleChangeCalendarMode = (date = null, modeIndex = 0) => {
+        this.refreshScreen()
+        this.setState({
+            calendarMode: ['month', 'day']?.[modeIndex], calendarTypeIndex: modeIndex,
+            nowYear: new Date(this.state?.selectedDate).getFullYear(),
+            nowMonth: new Date(this.state?.selectedDate).getMonth() + 1
+        })
+    }
+
+
+    getUsers = async () => {
```

Comment:
Is getUsers function used in reservation scenario?

Reply:
Now no use, but I see the data has a 'client' field, so I think maybe it's necessary